### PR TITLE
Remove APIs deprecated in 0.3.0.0.

### PR DESCRIPTION
**Pull request by patrickt**
Deletes `Eff`, `interpret`, and all the other functions that we
obsoleted with the carriers-as-monads proposal. This is
backwards-incompatible, but we provided adequate warning with the
previous release.

### CODE REVIEW

File: src/Control/Effect/Sum.hs [hunk 1]
```diff
@@ -55,7 +43,6 @@ instance {-# OVERLAPPABLE #-} Member sub sup => Member sub (sub' :+: sup) where
   prj (R g) = prj g
   prj _     = Nothing
 
-
```
Comment (robrix) [inline src/Control/Effect/Sum.hs]: I use two lines of whitespace to visually separate distinct chunks of the module; could we leave this in please? 🙏

Comment (patrickt) [reply src/Control/Effect/Sum.hs]: Oops, gotcha.

**Comment by patrickt** [comment]
Fixes #144.

**Comment by robrix** [comment]
Should we merge this before doing the 0.3.1 stuff (e.g. #137)? It’d mean existing users won’t get new toys without addressing usage of deprecated code; how big a deal is that going to be?

**Comment by patrickt** [comment]
@robrix I vote we merge this after 0.3.1, as a treat to ourselves.

**Comment by robrix** [comment]
🎉

**Pull request merged**